### ISSUE TITLE
sub commands like 'git remote prune origin'?

### ISSUE THREAD

**Issue by trusktr** [cacjs/cac#65]
## Issue Type

- [ ] Bug Report
- [ ] Feature Request
- [x] Other

There doesn't seem to be mention of this, so not sure if possible.

Example:

```
my-command remote rm origin
```

Maybe this is actually a feature request. I can change it if so.

**Comment by egoist** [comment]
Only 1-level sub command is currently supported, but I'm interested in this feature, do you know any library that actually supports this?

**Comment by trusktr** [comment]
Yeah, there are a few CLI tools with subcommands features, but they are all written plain JS:

- [yargs](https://www.npmjs.com/package/yargs)
  - example with folder hierarchy: https://github.com/yargs/yargs/blob/master/docs/advanced.md#example-command-hierarchy-using-commanddir
  - example with nested functions: https://github.com/yargs/yargs/issues/225#issuecomment-128532719
- [Sywac](http://sywac.io) is newer and inspired by Yargs (a bunch of parts of the documentation are WIP. @nexdrew)
  - Example with nested functions: http://sywac.io/docs/sync-config.html#command
  - Folder hierarchy: http://sywac.io/docs/sync-config.html#commandDirectory

**Comment by aleclarson** [comment]
Here's a lazy solution until @egoist figures out how to support this.

https://github.com/aleclarson/cac/commit/227835ee2a2621626c2fa75393787258287aab95

```ts
const app = cac()

app.command('foo bar').action(() => {})

// Parent command (if any) must come after child commands.
app.command('foo').action(() => {})
```

**Comment by tunnckoCore** [comment]
Well.. there's also another issue now.. :D Aliases doesn't work correctly.

**Comment by tunnckoCore** [comment]
@egoist want a PR?
